tests/test_output.py

```diff
@@ -0,0 +1,55 @@
+"""Test cases for confirming that generated docker compose files are valid."""
+
+import sys
+import unittest
+from os import unlink
+from subprocess import check_call
+from gen_docker_compose import DockerComposer
+from os.path import abspath, dirname, join, exists
+
+here = dirname(abspath(__file__))
+parent = dirname(here)
+templates_folder = join(here, 'data')
+template_file = 'docker-compose.template'
+rpi_output_file = join(here, 'data', 'rpi.yml')
+rockpi_output_file = join(here, 'data', 'rockpi.yml')
+
+sys.path.insert(1, parent)
+
+
+class TestValidComposeFileOutput(unittest.TestCase):
+
+    def setUp(self) -> None:
+        self._remove_output_files()
+        return super().setUp()
+
+    def tearDown(self) -> None:
+        self._remove_output_files()
+        return super().tearDown()
+
+    def _remove_output_files(self):
+        for filename in (rpi_output_file, rockpi_output_file):
+            if exists(filename):
+                unlink(filename)
+
+    def test_rpi_compose_output_is_valid(self):
```

Comment:
These tests are a little weak. They ideally would make an assertion about the contents of the generated file.

Reply:
Yes these were just initial versions to have at least confirmation that docker compose has no objections on the generated files. Some assertions I can think of:

1. Parse the generated yaml (using pyyaml?) and see that firmware version in services matches the one in settings.ini
2. Check that device arch and other parameters match those from settings.ini (though these will change once we start using them from hm-pyhelper).

I suggest if this is working now we update the test cases as part of #336 as we'll need to update the code and test cases for that any way.

Reply:
@kashifpk good idea - lets merge it into a single new issue / PR so we can close this out. it has been sat here for a while

Reply:
I agree with @kashifpk I think it would be best to test the generated file to ensure that settings.ini was applied correctly, we could continue doing it the way we are now. Where we generate files for the respective devices in a folder and then we could run tests on the .yml files in that folder.

Reply:
Updated AC for #336.